### ISSUE TITLE
Increase total asset limit for pgRawStreamer beyond 65535

### ISSUE THREAD

**Issue by Legacy-TacticalGamingInteractive** [citizenfx/fivem#2984]
### What happened?

`Increase total asset limit for pgRawStreamer beyond 65535`
 
Does anyone the FiveM team here know if this is still something in the future plans to get worked on?

A few months back some guys from the team were talking about it, as several servers have been hitting the limit. At that time we created the error that at least now shows clients when that limit is reached upon attempting to join a server that has reached the limit. Additionally one can check the limit with `assetscount` in console.

I am aware its no easy update so I'm told. However I am just wondering if it is still on the backlog of things for the future to improve FiveM.

We continually hits this limit and its difficult to simply remove stuff (especially awesome stuff you bought and love) from the server. Especially when everyone complains when stuff gets removed - yet also constantly wants new stuff. Understandably though, nobody wants to lose content of course. Now we are no longer really able to buy anything else on Tebex as much as we want to, as there is no point since the server is basically maxed out!

Thank you!

### Expected result

Increased pgRawStreamer limit

### Reproduction steps

Have a server streaming assets beyond the limit.

### Importancy

Crash

### Area(s)

FiveM, FXServer

### Specific version(s)

tx: v7.3.2 | fx: b11840 | gta build 3258

### Additional information

It would be a great service to the FiveM community. The ability to buy more content for servers and expand beyond the limit of streamed assets currently there would be incredible.

**Comment by ImBaphomettt** [comment]
Hello, I am experiencing the same issue. Indeed, it would be beneficial for some people to have this modification, especially with the increase in pool size, which now allows us to do more.

**Comment by Gltch3d** [comment]
I seems the ability to increase pool limits was somewhat counter intuitive. Correct me if I am wrong here but doing so doesn't increase the overall pgRawStreamer limit does it? That would likely further allow the issue to become a bigger one. The only real solution right now is to find any heavy resources having the most impact on your servers pgRawStreamer limit and remove it. I know it isnt the solution anyone is really wanting, but will be the only option until an update of some kind is considered if ever.

**Comment by ook3D** [comment]
> I seems the ability to increase pool limits was somewhat counter intuitive. Correct me if I am wrong here but doing so doesn't increase the overall pgRawStreamer limit does it? That would likely further allow the issue to become a bigger one. The only real solution right now is to find any heavy resources having the most impact on your servers pgRawStreamer limit and remove it. I know it isnt the solution anyone is really wanting, but will be the only option until an update of some kind is considered if ever.

How is the pool limit feature even relevant here, it's a completely different unrelated issue

**Comment by Gltch3d** [comment]
> > I seems the ability to increase pool limits was somewhat counter intuitive. Correct me if I am wrong here but doing so doesn't increase the overall pgRawStreamer limit does it? That would likely further allow the issue to become a bigger one. The only real solution right now is to find any heavy resources having the most impact on your servers pgRawStreamer limit and remove it. I know it isnt the solution anyone is really wanting, but will be the only option until an update of some kind is considered if ever.
> 
> How is the pool limit feature even relevant here, it's a completely different unrelated issue

What exactly are you talking about? The ability to increase pool size (which was added in a PR some time ago) allows you to increase the limit of specific asset types you can stream does it not? This would inherently allow someone add more txd as an example to the server only furthering the potential of reaching the overall pgRawStreamer limit.

**Comment by ook3D** [comment]
> > > I seems the ability to increase pool limits was somewhat counter intuitive. Correct me if I am wrong here but doing so doesn't increase the overall pgRawStreamer limit does it? That would likely further allow the issue to become a bigger one. The only real solution right now is to find any heavy resources having the most impact on your servers pgRawStreamer limit and remove it. I know it isnt the solution anyone is really wanting, but will be the only option until an update of some kind is considered if ever.
> > 
> > 
> > How is the pool limit feature even relevant here, it's a completely different unrelated issue
> 
> What exactly are you talking about? The ability to increase pool size (which was added in a PR some time ago) allows you to increase the limit of specific asset types you can stream does it not? This would inherently allow someone add more txd as an example to the server only furthering the potential of reaching the overall pgRawStreamer limit.

No not necessarily, pools are not only related to streamed assets, you can reach the streamer limit without any pool increases

**Comment by Gltch3d** [comment]
> > > > I seems the ability to increase pool limits was somewhat counter intuitive. Correct me if I am wrong here but doing so doesn't increase the overall pgRawStreamer limit does it? That would likely further allow the issue to become a bigger one. The only real solution right now is to find any heavy resources having the most impact on your servers pgRawStreamer limit and remove it. I know it isnt the solution anyone is really wanting, but will be the only option until an update of some kind is considered if ever.
> > > 
> > > 
> > > How is the pool limit feature even relevant here, it's a completely different unrelated issue
> > 
> > 
> > What exactly are you talking about? The ability to increase pool size (which was added in a PR some time ago) allows you to increase the limit of specific asset types you can stream does it not? This would inherently allow someone add more txd as an example to the server only furthering the potential of reaching the overall pgRawStreamer limit.
> 
> No not necessarily, pools are not only related to streamed assets, you can reach the streamer limit without any pool increases

Okay, that makes a little more sense. Thanks for the clarification.

**Comment by Legacy-TacticalGamingInteractive** [comment]
There is a max limit of all ydr, ytyp, ybn, ytd, ymt etc that can be streamed to clients. Unfortunately this causes issues on many servers especially long running servers who invest heavily in content from Tebex etc.

**Comment by ImBaphomettt** [comment]
Hello, @Legacy-TacticalGamingInteractive 

To be honest, I believe this issue will never be fixed, and we will have to adapt accordingly. If you're not familiar with it, allow me to introduce you to the following client command: `net_pgStats true.`

This command will help you track all the content you are streaming, easily remove duplicates, and most importantly, check how close you are to the limit.

Here is an example from my side:

![Image](https://github.com/user-attachments/assets/fce77e47-dc1b-43a5-8b29-eb4669e5f8d8)

**Comment by Legacy-TacticalGamingInteractive** [comment]
Well, perhaps, but the commit made by @blattersturm [https://github.com/citizenfx/fivem/commit/e5903edf7d5e2b7aff53723678c07f0db778ef65] might be possible to still refactor and update to work? I dont know. Theres some hope lol. I think @prikolium-cfx or someone had mentioned a few months ago that it was possible and was added to the backlog. But that was around 4 months ago now.

We've hit the limit so many times over while constantly removing good content and haven't been able to really buy anything on Tebex for our server in months now because whats the point when we are maxed out. Unfortunately.

**Comment by ImBaphomettt** [comment]
> Well, perhaps, but the commit made by [@blattersturm](https://github.com/blattersturm) [https://github.com/citizenfx/fivem/commit/e5903edf7d5e2b7aff53723678c07f0db778ef65] might be possible to still refactor and update to work? I dont know. Theres some hope lol. I think [@prikolium-cfx](https://github.com/prikolium-cfx) or someone had mentioned a few months ago that it was possible and was added to the backlog. But that was around 4 months ago now.
> 
> We've hit the limit so many times over while constantly removing good content and haven't been able to really buy anything on Tebex for our server in months now because whats the point when we are maxed out. Unfortunately.

I find myself in the exact same situation. It's unfortunate because, from a financial standpoint, this creates a significant loss of revenue for the CFX team.

If we no longer have the ability to integrate purchased content, we simply won’t buy it. I’ve had the opportunity to explore numerous servers, and unfortunately, I have to acknowledge that none of the major servers in my region fall below 60,000, except for three exceptions out of approximately 25 servers.

**Comment by Legacy-TacticalGamingInteractive** [comment]
INDEED, plus our members do not understand that there is a limit, even though we explain it to them. They still don't understand why we cannot put stuff in that they request, like new interiors or cars or clothing. They get so mad. Sometimes end up leaving to go to another server instead because of it. Sigh.

**Comment by VolCoM1337** [comment]
We're experiencing the same situation, even after optimizing our assets as much as possible. With large-scale projects like Gabz mappings, Gabz cars, Roxwood, or Las Venturas - some of which stream over 5,000 assets - this has become the norm for many servers.

I just want to emphasize how important this issue is for the community. Hopefully, this concern is being seriously considered, as it directly impacts the ability to run and expand high-quality servers.

**Comment by Legacy-TacticalGamingInteractive** [comment]
Indeed, but also many mappers don't know what they are doing - even some of the big ones (or they are lazy when it comes to the technical side). So, they often duplicate vanilla assets, even rename them (such as Roxwood project), rather than call them properly from a manifest. If you look at many of the map projects you will see thousands of props in some of them that are actually vanilla files, unedited, just renamed and escrow locked in their project. When all they had to actually do was call it properly in manifest files. It doesn't need to be re-streamed for no reason. But alas, that issue will continue.

**Comment by ook3D** [comment]
> 
> We've hit the limit so many times over while constantly removing good content and haven't been able to really buy anything on Tebex for our server in months now because whats the point when we are maxed out. Unfortunately.

I really have no clue how you have managed to do that, for prodigyRP we have loads of custom assets as well, mainly original stuff and have no issues, is it possible your clothing has too many texture options or models in general? as far as im aware, each asset counts towards the RawStreamer

**Comment by Legacy-TacticalGamingInteractive** [comment]
> > We've hit the limit so many times over while constantly removing good content and haven't been able to really buy anything on Tebex for our server in months now because whats the point when we are maxed out. Unfortunately.
> 
> I really have no clue how you have managed to do that, for prodigyRP we have loads of custom assets as well, mainly original stuff and have no issues, is it possible your clothing has too many texture options or models in general? as far as im aware, each asset counts towards the RawStreamer

We've been going through everything we have constantly. We just have a lot of content. 

Re: The post I made above is mostly the reason. Mappers and devs who don't care about optimization or file structure because once they have your money, they have your money lol. Most dont even run servers themselves so what do they care. With things being escrowed as well it just adds to the problem otherwise things could be condensed even more.

Any ytd, ydr, yft, ymt, ymf, ycd, ynv, ybn etc count towards the limit.

Not to mention peoples client sided mods they stream (but thats on them, we usually tell them to remove a bunch of their mods because it stops them from even joining if they have too many when the limit is close of course).

**Comment by ImBaphomettt** [comment]
A fix to the problem is currently available in experiemental. I think this issue can be closed as soon as possible.

https://forum.cfx.re/t/pgrawstreamer-limits/5305317/3

**Comment by Gltch3d** [comment]
> A fix to the problem is currently available in experiemental. I think this issue can be closed as soon as possible.
> 
> https://forum.cfx.re/t/pgrawstreamer-limits/5305317/3

page doesnt exist, unless it's permission locked

**Comment by DaniGP17** [comment]
> page doesnt exist, unless it's permission locked

Is only available for members inside the experimental group of the forum

**Comment by Gltch3d** [comment]
So this error theoretically (assuming by the information only provided in this group suggesting it's only being tested) would be non existent when using the experiential branch for the FiveM client? Doesn't necessarily mean this thread should be closed IMO since requesting players to use that branch could cause other issues since there's a chance that the branch could become unusable at times . Very existing info nonetheless.

**Comment by prikolium-cfx** [comment]
> So this error theoretically (assuming by the information only provided in this group suggesting it's only being tested) would be non existent when using the experiential branch for the FiveM client? Doesn't necessarily mean this thread should be closed IMO since requesting players to use that branch could cause other issues since there's a chance that the branch could become unusable at times . Very exciting info nonetheless.

If there won't be any issues it will be push to canary in next few days.

**Comment by Gltch3d** [comment]
> > So this error theoretically (assuming by the information only provided in this group suggesting it's only being tested) would be non existent when using the experiential branch for the FiveM client? Doesn't necessarily mean this thread should be closed IMO since requesting players to use that branch could cause other issues since there's a chance that the branch could become unusable at times . Very exciting info nonetheless.
> 
> If there won't be any issues it will be push to canary in next few days.

That's awesome, great work

**Comment by Legacy-TacticalGamingInteractive** [comment]
> > So this error theoretically (assuming by the information only provided in this group suggesting it's only being tested) would be non existent when using the experiential branch for the FiveM client? Doesn't necessarily mean this thread should be closed IMO since requesting players to use that branch could cause other issues since there's a chance that the branch could become unusable at times . Very exciting info nonetheless.
> 
> If there won't be any issues it will be push to canary in next few days.

❤️❤️❤️❤️💪🏼

**Comment by prikolium-cfx** [comment]
In this 1d078e3c4c51e8e12be546d56a03cbfa70c2212f commit we increased that limit, so now you can load 65k ytds and 65k of non ytds different assets.

**Comment by PeekerJ** [comment]
> In this [1d078e3](https://github.com/citizenfx/fivem/commit/1d078e3c4c51e8e12be546d56a03cbfa70c2212f) commit we increased that limit, so now you can load 65k ytds and 65k of non ytds different assets.

Great, crashes now occur everytime when launching FiveM

![Image](https://github.com/user-attachments/assets/3e7b9bcd-42e4-4b39-92fe-ff4a77415460)

**Comment by prikolium-cfx** [comment]
> > In this [1d078e3](https://github.com/citizenfx/fivem/commit/1d078e3c4c51e8e12be546d56a03cbfa70c2212f) commit we increased that limit, so now you can load 65k ytds and 65k of non ytds different assets.
> 
> Great, crashes now occur everytime when launching FiveM
> 
> ![Image](https://github.com/user-attachments/assets/3e7b9bcd-42e4-4b39-92fe-ff4a77415460)

Please provide information. What assets are you using, which server connecting

**Comment by PeekerJ** [comment]
> > > In this [1d078e3](https://github.com/citizenfx/fivem/commit/1d078e3c4c51e8e12be546d56a03cbfa70c2212f) commit we increased that limit, so now you can load 65k ytds and 65k of non ytds different assets.
> > 
> > 
> > Great, crashes now occur everytime when launching FiveM
> > ![Image](https://github.com/user-attachments/assets/3e7b9bcd-42e4-4b39-92fe-ff4a77415460)
> 
> Please provide information. What assets are you using, which server connecting

https://servers.fivem.net/servers/detail/7dl4qr Here are all the assets in the server, the configuration isn't as different, the only thing it could be are the "increase_pool_size" in the server.cfg:

increase_pool_size "TxdStore" 26000
increase_pool_size "CMoveObject" 75
increase_pool_size "netGameEvent" 400
increase_pool_size "StaticBounds" 5000

The issue started appearing on latest after the commit, even when just clicking the connect button FiveM crashes instantly.

**Comment by prikolium-cfx** [comment]
> > > > In this [1d078e3](https://github.com/citizenfx/fivem/commit/1d078e3c4c51e8e12be546d56a03cbfa70c2212f) commit we increased that limit, so now you can load 65k ytds and 65k of non ytds different assets.
> > > 
> > > 
> > > Great, crashes now occur everytime when launching FiveM
> > > ![Image](https://github.com/user-attachments/assets/3e7b9bcd-42e4-4b39-92fe-ff4a77415460)
> > 
> > 
> > Please provide information. What assets are you using, which server connecting
> 
> https://servers.fivem.net/servers/detail/7dl4qr Here are all the assets in the server, the configuration isn't as different, the only thing it could be are the "increase_pool_size" in the server.cfg:
> 
> increase_pool_size "TxdStore" 26000 increase_pool_size "CMoveObject" 75 increase_pool_size "netGameEvent" 400 increase_pool_size "StaticBounds" 5000
> 
> The issue started appearing on latest after the commit, even when just clicking the connect button FiveM crashes instantly.

Thanks for the information provided. We are investigating this issue right now

**Comment by PeekerJ** [comment]
> > > > > In this [1d078e3](https://github.com/citizenfx/fivem/commit/1d078e3c4c51e8e12be546d56a03cbfa70c2212f) commit we increased that limit, so now you can load 65k ytds and 65k of non ytds different assets.
> > > > 
> > > > 
> > > > Great, crashes now occur everytime when launching FiveM
> > > > ![Image](https://github.com/user-attachments/assets/3e7b9bcd-42e4-4b39-92fe-ff4a77415460)
> > > 
> > > 
> > > Please provide information. What assets are you using, which server connecting
> > 
> > 
> > https://servers.fivem.net/servers/detail/7dl4qr Here are all the assets in the server, the configuration isn't as different, the only thing it could be are the "increase_pool_size" in the server.cfg:
> > increase_pool_size "TxdStore" 26000 increase_pool_size "CMoveObject" 75 increase_pool_size "netGameEvent" 400 increase_pool_size "StaticBounds" 5000
> > The issue started appearing on latest after the commit, even when just clicking the connect button FiveM crashes instantly.
> 
> Thanks for the information provided. We are investigating this issue right now

I've managed to fix the issue by removing the following plugins from the folder:

HeapAdjuster.asi
HeapAdjuster.ini
PackfileLimitAdjuster.asi
PackfileLimitAdjuster.ini

**Comment by PeekerJ** [comment]
Would it be possible to increase the Building Pool size above 65000 too? @prikolium-cfx

**Comment by Legacy-TacticalGamingInteractive** [comment]
He may ask you to open a new issue for that. As its a different problem.

**Comment by prikolium-cfx** [comment]
> Would it be possible to increase the Building Pool size above 65000 too? [@prikolium-cfx](https://github.com/prikolium-cfx)



> He may ask you to open a new issue for that. As its a different problem.


No need, I extended that limit by 20k, you can try in ~1 hour to use pool resize feature.

**Comment by ImBaphomettt** [comment]
> > Would it be possible to increase the Building Pool size above 65000 too? [@prikolium-cfx](https://github.com/prikolium-cfx)
> 
> > He may ask you to open a new issue for that. As its a different problem.
> 
> No need, I extended that limit by 20k, you can try in ~1 hour to use pool resize feature.


Sorry to continue the discussion here, but since PeekerJ mentioned an increase in the pool size, I’m sending this so you can take a look and respond, whether positively or negatively. :) https://github.com/citizenfx/fivem/issues/2992

**Comment by PeekerJ** [comment]
> > Would it be possible to increase the Building Pool size above 65000 too? [@prikolium-cfx](https://github.com/prikolium-cfx)
> 
> > He may ask you to open a new issue for that. As its a different problem.
> 
> No need, I extended that limit by 20k, you can try in ~1 hour to use pool resize feature.

Great! Thank you! :D

**Comment by huncho2turnt** [comment]
Where is the file to download for this fix
